src/main/java/ui/MenuControl.java

```diff
@@ -91,59 +89,82 @@ private Menu createFileMenu() {
     }
 
     /**
-     * Logs the user action and creates a new board
+     * Checks whether the current board is dirty, i.e. has unsaved changes.
+     *
+     * @return true if current board is dirty, false otherwise
      */
-    private void onBoardNew() {
-        logger.info("Menu: Boards > New");
-        newBoard();
+    public final boolean isCurrentBoardDirty() {
+        if (!prefs.getLastOpenBoardPanelInfos().isPresent()) {
+            return true;
+        }
+        return !panels.getCurrentPanelInfos().equals(prefs.getLastOpenBoardPanelInfos().get());
     }
 
     /**
-     * Creates an empty board. User will be prompted to
-     * confirm the action if there are unclosed panels
+     * Prompts a dialog to ask the user whether or not to save the current board.
```

Comment:
The method is doing more than just prompting. Both method name and comment misleading.